File: common/src/main/java/cn/rubintry/common/utils/logger/Logger.java
package cn.rubintry.common.utils.logger;

import android.util.Log;

/**
 * @author logcat
 */
public class Logger implements ILogger {
    private String TAG;

    public Logger(String TAG) {
        this.TAG = TAG;
    }

    @Override
    public void debug(String message) {
        Log.d(TAG, message);
    }

    @Override
    public void debug(String message, Throwable ex) {
        Log.d(TAG, message , ex);
    }

    @Override
    public void error(String message) {
        Log.e(TAG, message);
    }

    @Override
    public void error(String message, Throwable ex) {
        Log.e(TAG, message , ex);
    }

    @Override
    public void warn(String message) {
        Log.w(TAG, message);
    }

    @Override
    public void warn(String message, Throwable ex) {
        Log.w(TAG, message , ex);
    }

    @Override
    public void verbose(String message) {
        Log.v(TAG , message);
    }

    @Override
    public void verbose(String message, Throwable ex) {
        Log.v(TAG , message , ex);
    }
}

File: lib-common/src/main/java/cn/rubintry/lib_common/thread/Logger.kt
package cn.rubintry.lib_common.thread

import android.util.Log

object Logger {
    private val TAG = this.javaClass.simpleName
    private var open = false

    @JvmStatic
    fun open(){
        open = true
    }

    @JvmStatic
    fun debug(message: String) {
        if(open){
            Log.d(TAG, message)
        }
    }


    @JvmStatic
    fun error(message: String) {
        if(open){
            Log.e(TAG, message)
        }
    }

    @JvmStatic
    fun error(message: String, throwable: Throwable) {
        if(open){
            Log.e(TAG, message, throwable)
        }
    }

    @JvmStatic
    fun error(throwable: Throwable){
        if(open){
            Log.e(TAG, "" , throwable)
        }
    }

    @JvmStatic
    fun info(message: String) {
        if(open){
            Log.i(TAG, message)
        }
    }


    @JvmStatic
    fun warn(message: String) {
       if(open){
           Log.w(TAG, message)
       }
    }


}